src/CarbonAware.WebApi/test/integrationTests/SciScoreControllerTests.cs

```diff
@@ -0,0 +1,76 @@
+namespace CarbonAware.WepApi.IntegrationTests;
+
+using CarbonAware.DataSources.Configuration;
+using CarbonAware.WebApi.IntegrationTests;
+using NUnit.Framework;
+using System.Net;
+using System.Net.Http.Headers;
+using System.Net.Mime;
+using System.Text.Json;
+
+/// <summary>
+/// Tests that the Web API controller handles and packages various responses from a plugin properly 
+/// including empty responses and exceptions.
+/// </summary>
+[TestFixture(DataSourceType.JSON)]
+[TestFixture(DataSourceType.WattTime)]
+public class SciScoreControllerTests : IntegrationTestingBase
+{
+    private string marginalCarbonIntensityURI = "/sci-scores/marginal-carbon-intensity";
+    public SciScoreControllerTests(DataSourceType dataSource) : base(dataSource) { }
+
+    [TestCase("2022-1-1", 1, "eastus")]
+    public async Task SCI_WithValidData_ReturnsContent(DateTime start, int offset, string location)
+    {
+        var end = start.AddDays(offset);
+        _dataSourceMocker.SetupDataMock(start, end, location);
+        string timeInterval = start.ToUniversalTime().ToString("O") + "/" + end.ToUniversalTime().ToString("O");
+
+        //Construct body object and then serialize it with JSONSerializer
+        object body = new
+        {
+            location = new
+            {
+                locationType = "CloudProvider",
+                providerName = "Azure",
+                regionName = location
+            },
+            timeInterval = timeInterval
+        };
+
+        var jsonBody = JsonSerializer.Serialize(body);
+        StringContent _content = new StringContent(jsonBody);
+
+        var mediaType = new MediaTypeHeaderValue(MediaTypeNames.Application.Json);
+        _content.Headers.ContentType = mediaType;
+
+        var result = await _client.PostAsync(marginalCarbonIntensityURI, _content);
```

Comment:
nit: It feels like these lines (41..48) are the same, so maybe invest on a common method to be shared between these 2 tests.

Reply:
Done.